components/server/src/workspace/workspace-starter.ts

```diff
@@ -349,6 +351,20 @@ export class WorkspaceStarter {
         }
     }
 
+    public async stopWorkspaceInstance(
+        ctx: TraceContext,
+        instanceId: string,
+        instanceRegion: string,
+        policy?: StopWorkspacePolicy,
+    ): Promise<void> {
+        const req = new StopWorkspaceRequest();
+        req.setId(instanceId);
+        req.setPolicy(policy || StopWorkspacePolicy.NORMALLY);
+
+        const client = await this.clientProvider.get(instanceRegion);
```

Comment:
If I remember correctly, some instances might not have a region yet (e.g. still in image build or so?) and this could cause `client` to be incorrect or undefined. Is this still the case? 🤔 I guess this needs to be tested.